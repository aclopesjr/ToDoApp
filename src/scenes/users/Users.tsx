import React, {useState, useEffect} from 'react';
import {StyleSheet, FlatList, View, RefreshControl} from 'react-native';
import {User} from './components';
import {fetchUsersAsync} from '../../service/connector';
import {IUser} from '../../types';

const Users = (props: any) => {
  const [users, setUsers] = useState(Array<IUser>());
  const [isRefresing, setRefresh] = useState(false);

  const _keyExtractor = (item: IUser) => item.id.toString();

  const _fetchDataAsync = async () => {
    setRefresh(true);
    const users = await fetchUsersAsync();
    setUsers(users);
    setRefresh(false);
  };

  const _renderItem = ({item}: {item: IUser}) => {
    return (
      <User
        name={item.name}
        email={item.email}
        username={item.username}
        onPress={() => props.navigation.navigate('Tasks', {user: item})}
      />
    );
  };

  useEffect(() => {
    _fetchDataAsync();
  }, []);

  return (
    <FlatList
      data={users}
      refreshing={isRefresing}
      refreshControl={
        <RefreshControl
          refreshing={isRefresing}
          onRefresh={async () => await _fetchDataAsync()}
        />
      }
      extraData={props}
      keyExtractor={_keyExtractor}
      renderItem={_renderItem}
      ItemSeparatorComponent={() => <View style={styles.separator} />}
    />
  );
};

export default Users;

const styles = StyleSheet.create({
  separator: {
    borderBottomWidth: 1,
    borderColor: 'lightgray',
    marginLeft: 12,
    marginRight: 12,
  },
});
