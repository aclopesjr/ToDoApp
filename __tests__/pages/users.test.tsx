import React from 'react';
import {shallow} from 'enzyme';
import Users from '../../src/scenes/users/Users';

describe('Snapshot Users Page', () => {
  it('Renders Users correctly', () => {
    const component = shallow(<Users />);
    expect(component).toMatchSnapshot();
  });
});
