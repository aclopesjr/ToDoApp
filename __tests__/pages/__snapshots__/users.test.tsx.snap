// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Users Page Renders Users correctly 1`] = `ShallowWrapper {}`;
